features/unattached_status.feature

```diff
@@ -40,155 +40,389 @@ Feature: Unattached status
+        """
+        SERVICE         +AVAILABLE +DESCRIPTION
+        esm-apps        +yes       +Extended Security Maintenance for Applications
+        esm-infra       +yes       +Extended Security Maintenance for Infrastructure
+        fips            +yes       +NIST-certified core packages
+        fips-updates    +yes       +NIST-certified core packages with priority security updates
+        livepatch       +yes       +Canonical Livepatch service
+        usg             +yes       +Security compliance and audit tools
+
+        This machine is not attached to an Ubuntu Pro subscription.
+        See https://ubuntu.com/pro
+        """
+        When I run `pro status --all` with sudo
+        Then stdout matches regexp:
+        """
+        SERVICE         +AVAILABLE +DESCRIPTION
+        cc-eal          +no        +Common Criteria EAL2 Provisioning Packages
+        esm-apps        +yes       +Extended Security Maintenance for Applications
+        esm-infra       +yes       +Extended Security Maintenance for Infrastructure
+        fips            +yes       +NIST-certified core packages
+        fips-updates    +yes       +NIST-certified core packages with priority security updates
+        livepatch       +yes       +Canonical Livepatch service
+        realtime-kernel +no        +Beta-version Ubuntu Kernel with PREEMPT_RT patches
+        ros             +no        +Security Updates for the Robot Operating System
+        ros-updates     +no        +All Updates for the Robot Operating System
+        usg             +yes       +Security compliance and audit tools
+
+        This machine is not attached to an Ubuntu Pro subscription.
+        See https://ubuntu.com/pro
+        """ 
+        When I append the following on uaclient config:
+        """
+        features:
+            allow_beta: true
+        """
+        And I run `pro status` as non-root
+        Then stdout matches regexp:
+        """
+        SERVICE         +AVAILABLE +DESCRIPTION
+        esm-apps        +yes       +Extended Security Maintenance for Applications
+        esm-infra       +yes       +Extended Security Maintenance for Infrastructure
+        fips            +yes       +NIST-certified core packages
+        fips-updates    +yes       +NIST-certified core packages with priority security updates
+        livepatch       +yes       +Canonical Livepatch service
+        usg             +yes       +Security compliance and audit tools
+
+        FEATURES
+        allow_beta: True
+
+        This machine is not attached to an Ubuntu Pro subscription.
+        See https://ubuntu.com/pro
+        """ 
+
+        Examples: ubuntu release
+           | release |
+           | focal   |
+
+    @series.jammy
+    @uses.config.machine_type.lxd.container
+    Scenario Outline: Unattached status in a ubuntu machine
+        Given a `<release>` machine with ubuntu-advantage-tools installed
+        When I run `pro status` as non-root
```

Comment:
I appreciate the thoroughness of these additions! And like we talked about earlier, I think we should extend this idea to all machine_types x all series, but thats for a future PR.

I think we can simplify it in one way though. Which is that non-root vs root status should always be the same when unattached right? (and after #2085 they will always be the same when attached)

So I think there should maybe be one test that verifies they are the same, but I don't think we need both for every series - what do you think?

Reply:
We could create a step that checks if they are equal and use only `non-root` calls for the tests. What do you think ?

Reply:
We can add it to the current run step, extending `as non-root` and `with sudo` with another option, who would run the command as both and make sure it is the same I guess.

Reply:
I like the idea of a new step, maybe something like:
```
When I run ua status `--flags --go --here`
```
That runs the command as both root and non-root, asserts they have equal stdout,stderr, and then sets the appropriate stdout/stderr variables on the behave context

Reply:
I think I would prefer something like `When I verify root and non-root status `flags` calls have the same output`. The idea is just to be as explicit as possible. But let me know what you all think

Reply:
That works for me!